airbytehq/airbyte-python-cdk
# fix: ensure SDM image maintains airbyte-ci compatibility

## What

When we implemented the new publish flow for SDM in Dockerhub, the built image diverged from the existing structure of source-declarative-manifest images built via the `base_images` package in our monorepo. This had the unintentional side effect of breaking existing airbyte-ci commands for manifest-only connectors, which have set expectations about the image's entrypoint and directory structure.

## Discrepancy

![Screenshot 2024-11-21 at 4 47 33 PM](https://github.com/user-attachments/assets/3cc5990c-e378-42f1-8336-4e8d19bde3c7)

## How

This PR modifies the built Dockerfile image to resemble the structure of the original SDM image by adding a `main.py` file at the project's root level and renaming `_run.py` to `run.py` within the built image.

## Testing Steps Used

1. Published SDM version `6.7.0rc5` using the updated Dockerfile in this branch
1. Pulled the images for SDM versions `6.5.2`(stable monorepo release) and `6.7.0rc5` from Dockerhub
3. Inspected the images using `docker run --rm -it --entrypoint /bin/bash source-declarative-manifest:<version>` for comparison
4. Set the base image for source-pokeapi locally to `6.7.0rc5`
5. Ran `airbyte-ci connectors --name=source-pokeapi build` to verify the image builds locally
6. Ran all of the connector interfaces against the local image to verify the connector works as expected.
7. Created a cusom PokeAPI source in Airbyte Cloud and pinned the version of SDM to `6.7.0rc5` to verify the connector syncs on the platform
